core/types/src/keys.rs

```diff
@@ -0,0 +1,356 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Basic key types
+
+use core::{
+    fmt::{Debug, Display},
+    marker::PhantomData,
+};
+
+use curve25519_dalek::{ristretto::RistrettoPoint, scalar::Scalar};
+use zeroize::Zeroize;
+
+use mc_crypto_keys::{KeyError, RistrettoPrivate, RistrettoPublic};
+
+// Exported key types
+
+/// Subaddress view private key
+pub type SubaddrViewPrivate = Key<Subaddr, View, RistrettoPrivate>;
```

Comment:
:wrench: The use of `Subaddr` here  is asymetrical with that of `Subaddress` in `accounts.rs` and `subaddress.rs` suggest spelling it out here as well.

Reply:
for the type aliases to work the label types have to be public, so it's actually sort of deliberate that the labels are abbreviated so they don't collide with other names.

another approach i have used in the past is to be more explicit in the labels so it's like `SubaddressTy`, could also do that if it's preferred?

Reply:
blarg....

It may be that I'm bringing baggage of naming gone bad.  My hunch is someone will cargo cult the `Subaddr` and not realize that it was OK because it was a marker trait.

I was able to:
- create a new module `markers.rs`
-  move the marker types into `markers.rs`
- add `mod markers` to `lib.rs`
- add `use crate::markers::{Subaddr, View, Spend, Root};` to `keys.rs`
- re-run the subaddress tests in `mc-core`

Not sure if there is another use case needed here where the marker traits are desired outside the `mc-core-types` crate